openproblems/tasks/cell_cell_communication/README.md

```diff
@@ -0,0 +1,72 @@
+# Cell-cell Communication
+
+## The task
+
+The growing availability of single-cell data has sparked an increased
+interest in the inference of cell-cell communication (CCC),
+with an ever-growing number of computational tools developed for this purpose.
+
+Different tools propose distinct preprocessing steps with diverse
+scoring functions, that are challenging to compare and evaluate.
+Furthermore, each tool typically comes with its own set of prior knowledge.
+To harmonize these, we recently developed the
+[LIANA](https://github.com/saezlab/liana) framework, which is
+used extensively at the current stage of the task.
+
+The challenges in evaluating the tools are further exacerbated by the
+lack of an appropriate gold standard to benchmark the performance of
+CCC methods. In a [recent publication](https://rdcu.be/cR69y), we used
+alternative data modalities, including the spatial proximity of cell types
+and inferred downstream cytokine activities, to assess the ability
+of the methods to detect biological signal. Yet, these modalities are only
+approximations of biological reality and come with their own assumptions
+and limitation. We thus hope that the **OpenProblems** platform will allow
+for the inclusion of more datasets with known ground truth interactions,
+with which the limitations and advantages of the
+different CCC methods will be better understood.
+
+## The metrics
+
+Metrics for cell-cell communication aim to characterize how good are
+the different scoring functions/methods at prioritizing
+assumed truth predictions.
+
+`Odds ratios`: represent the ratio of true and false positives within a set of
+prioritized interactions (top ranked hits) versus the same ratio for the
+remainder of the interactions. Thus, in this scenario odds ratios quantify
+the strength of association between the ability of methods to prioritize
+interactions and those interactions assigned to the positive class.
+
+## API
+
+Datasets should include cell type annotations in `adata.obs["label"]`,
+along with some assumed truth in `adata.uns["bench"]` with a binary
+`response` vector [0; 1], and the corresponding columns relevant for the
+benchmark. The assumed truth could essentially be in
+different forms, and we refer the reader to [our recent publication for more
+details](https://rdcu.be/cSs92).
+
+For example, the triple negative breast cancer dataset (`tnbc_wu2021`) portrays
```

Comment:
My concern here is how the methods are expected to handle the differing API. Sometimes `ligand` and `target` are available, sometimes they are not? Those sound like different subtasks, unless I am misunderstanding. cc @LuckyMD

Reply:
Each of the methods are expected to have `ligand`, `source`, and  `target`. The assumed truth can thus be assigned to either combination of those columns, so the API stays the same, but the assumed truth can be flexible.

Reply:
Perhaps, I could clarify this further in the README. 

Technically, all of the currently-implemented methods return 5 columns: `ligand`, `receptor`, `source`, `target`, and `score`. However, since `receptor` is not used in either of the benchmarks, I have not enforced it. This way we are potentially also allowing for methods which focus on ligands alone (e.g. a modified version NicheNet) to be included.

Being as flexible to the type of truth that one could use to benchmark is IMHO essential. For example, different technologies might be of interest:

- Physically-interacting cell type pairs (relevant columns" `source`, `target`)
- Tagged/labelled ligands (relevant columns: `source`, `target`, `ligand`)
etc

Reply:
My concern here is that metrics have to be able to operate on all input datasets, so they need to know what the input will look like. Can we be more specific about what columns are expected to be included and then allow some to be null? e.g.

> `adata.uns["ccc_target"]` should be a pandas DataFrame containing the following columns:
> * `response`: `int`, binary response variable indicating whether or not an interaction is assumed to have occurred
> * `source`: `str`, named of source cell type in interaction
> * `target`: `str`, named of target cell type in interaction
> * `ligand`: `str`, gene symbol of the relevant ligand in interaction (may be null)
> * `receptor`: `str`, gene symbol of the relevant ligand receptor in interaction (may be null)
> * `score`: `float`, score between 0 and 1 giving an inferred strength of the assumed interaction

Reply:
Hi Scott,

Great suggestion and thanks for the changes to the README.
Have now wrote it to be much more explicit as suggested by yourself.